backend/src/main/java/com/javahelp/backend/domain/user/register/client/IClientRegisterInputBoundary.java

```diff
@@ -0,0 +1,32 @@
+package com.javahelp.backend.domain.user.register.client;
+
+import com.javahelp.backend.domain.user.register.IUserRegisterInputBoundary;
+import com.javahelp.model.user.UserType;
+
+public interface IClientRegisterInputBoundary extends IUserRegisterInputBoundary {
+
+    /**
+     * @return the living address from the (Client/Provider) RequestModel
+     */
+    String getAddress();
+
+    /**
+     * @return the phone number from the (Client/Provider) RequestModel
+     */
+    String getPhoneNumber();
+
+    /**
+     * @return first name from the ClientRequestModel
+     */
+    String getFirstName();
+
+    /**
+     * @return last name from the ClientRequestModel
+     */
+    String getLastName();
+
+    /**
+     * @return the type of that {@link com.javahelp.model.user.User}
+     */
+    UserType getType();
```

Comment:
Every `UserInfo` coming out of here will have the same `UserType` so this is unnecessary.

Reply:
Agree, though we also agreed on keeping it for future usage.

Reply:
It should not be in either, the subclass of `UserRegisterInteractor` used should only process a single type of `User` so this is not necessary anywhere.

Reply:
Gotcha